### ISSUE TITLE
NullReferenceException in ClusterConnections.OnMembersUpdated when smart routing is disabled

### ISSUE THREAD

**Issue by drieseng** [hazelcast/hazelcast-csharp-client#465]
When smart routing is disabled in version **4.0.1**, the following (non-fatal) exception is reported when the connection is established (and logging is enabled):

```txt
fail: Hazelcast.Clustering.DistributedEventScheduler[0]
      An event handler [2] has thrown an unhandled exception.
      System.NullReferenceException: Object reference not set to an instance of an object.
         at Hazelcast.Clustering.ClusterConnections.OnMembersUpdated(MembersUpdatedEventArgs args)
         at Hazelcast.Core.FuncExtensions.<AwaitEach>d__1`1.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Hazelcast.Clustering.ClusterEvents.<HandleCodecMemberViewEvent>d__36.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Hazelcast.Clustering.ClusterSubscription.<HandleAsync>d__30.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Hazelcast.Clustering.DistributedEventScheduler.<Handle>d__21.MoveNext()
```

I'll submit a PR to fix this issue.

**Comment by zpqrtbnk** [comment]
Hey - I can confirm the issue. Yet, don't rush on a PR, as the issue should be fixed with upcoming version 4.0.2. 

As much as we love contributions, I don't want you to waste precious time.
How is your experience with this client so far?

**Comment by drieseng** [comment]
We had some issue (on our end) with the "async all-the-way" approach, but it's looking good now.
Documentation isn't always on par, but we all struggle with this :-)

Do I just close this issue (in light of the upcoming 4.0.2 release)?

**Comment by zpqrtbnk** [comment]
Yea... doc is lagging but working on it.
Feel free to ask questions here re. async. It's *good* for perfs, but can be... confusing at times.
Don't close the issue yet - we'll close when 4.0.2 is out.

**Comment by zpqrtbnk** [comment]
4.0.2 is out now, closing the issue as I don't expect the exception to show again (was aware of it and fixed it in 4.0.2)
